mypy_django_plugin/transformers/managers.py

```diff
@@ -274,6 +285,20 @@ def resolve_manager_method(ctx: AttributeContext) -> MypyType:
 
     if isinstance(ctx.type, Instance):
         return resolve_manager_method_from_instance(instance=ctx.type, method_name=method_name, ctx=ctx)
+    elif isinstance(ctx.type, UnionType):
+        if not all(isinstance(instance, Instance) for instance in ctx.type.items):
+            ctx.api.fail(
+                f'Unable to resolve return type of queryset/manager method "{method_name}"',
+                ctx.context,
+            )
+            return AnyType(TypeOfAny.from_error)
+
+        resolved = tuple(
+            resolve_manager_method_from_instance(instance=instance, method_name=method_name, ctx=ctx)
+            for instance in ctx.type.items
+            if isinstance(instance, Instance)
```

Comment:
mypy doesn't know about the all() check above and we have to reassure it these are indeed Instance objects

Reply:
What if we apply this diff?

```diff
-    elif isinstance(ctx.type, UnionType):
-        if not all(isinstance(instance, Instance) for instance in ctx.type.items):
-            ctx.api.fail(
-                f'Unable to resolve return type of queryset/manager method "{method_name}"',
-                ctx.context,
-            )
-            return AnyType(TypeOfAny.from_error)
-
+    elif isinstance(ctx.type, UnionType) and all(isinstance(instance, Instance) for instance in ctx.type.items):
         resolved = tuple(
             resolve_manager_method_from_instance(instance=instance, method_name=method_name, ctx=ctx)
             for instance in ctx.type.items
-            if isinstance(instance, Instance)
         )
         return get_proper_type(UnionType(resolved))
-    else:
    ctx.api.fail(f'Unable to resolve return type of queryset/manager method "{method_name}"', ctx.context)
    return AnyType(TypeOfAny.from_error)
```

Reply:
it still needs that extra isinstance in that case, but is definitely still an improvement :)